sunpy/net/hek/utils.py

```diff
@@ -0,0 +1,151 @@
+                else:
+                    new_column.append(value * get_unit(attribute["unit_prop"], table[unit_attr][idx], is_coord_prop = is_coord_prop))
+            table[attribute["name"]] = new_column
+
+    for attribute in attributes:
+        if attribute.get("is_unit_prop", False) and attribute["name"] in table.colnames:
+            del table[attribute["name"]]
+    return table
+
+# NOTE: Needs unit test
+def parse_unit(table, attribute, is_coord_prop = False):
+    if attribute.get("is_chaincode", False):
+        return table
+    unit_attr = ""
+    if is_coord_prop:
+        unit_attr = "event_coordunit"
+    else:
+        unit_attr = attribute["unit_prop"]
+    for row in table:
+        if unit_attr in table.colnames and row[unit_attr] not in ["", None] and table[attribute["name"]].unit is not None:
+            table[attribute["name"]].unit = get_unit(attribute["unit_prop"], row[unit_attr], is_coord_prop = is_coord_prop)
+            break
+    return table
+
+# NOTE: Needs unit test
+def parse_chaincode(value, idx, attribute, unit_prop):
+    coord1_unit = u.deg
+    coord2_unit = u.deg
+    if attribute["frame"] == "helioprojective":
+        coord1_unit = u.arcsec
+        coord2_unit = u.arcsec
+    elif attribute["frame"] == "heliocentric":
+        coord1_unit = u.R_sun
+        coord2_unit = u.deg
+    elif attribute["frame"] == "icrs":
+        coord1_unit = get_unit("coord1_unit", unit_prop, is_coord_prop = True)
+        coord2_unit = get_unit("coord2_unit", unit_prop, is_coord_prop = True)
+
+    coordinates_str = value.split('((')[1].split('))')[0]
+    coord1_list = [float(coord.split()[0]) for coord in coordinates_str.split(',')] * coord1_unit
+    coord2_list = [float(coord.split()[1]) for coord in coordinates_str.split(',')] * coord2_unit
+    vertices = {}
+    if attribute["frame"] == "heliocentric":
+       vertices = SkyCoord(coord1_list, coord2_list, [1]* len(coord1_list)* u.AU, representation_type = "cylindrical", frame = "heliocentric")
+    else:
+        vertices = SkyCoord(coord1_list, coord2_list, frame=attribute["frame"])
+
+    return PolygonSkyRegion(vertices = vertices)
+
+# NOTE: Needs unit test
+def get_unit(unit_prop, str, is_coord_prop = False):
+    """Converts str into astropy unit."""
+    cm2 = u.def_unit("cm2", u.cm**3)
+    m2 = u.def_unit("m2", u.m**2)
+    m3 = u.def_unit("m3", u.m**3)
+
+    aliases = {
+        "steradian": u.sr,
+        "arcseconds": u.arcsec,
+        "degrees": u.deg,
+        "sec": u.s,
+        "Emx": u.Mx,
+        "Amperes": u.A,
+        "ergs": u.erg,
+        "cubic centimeter": u.ml,
+        "square centimeter": cm2,
+        "cubic meter": m3,
+        "square meter": m2,
+    }
+
+    with u.add_enabled_units([cm2, m2, m3]), u.set_enabled_aliases(aliases):
+        if is_coord_prop:
+            coord1_unit, coord2_unit, coord3_unit = None, None, None
+            coord_units = re.split(r'[, ]', str)
```

Comment:
Do you know what units would split here?

Reply:
The coordination units, right?
For example arcsec,arcsec

Reply:
Ah I see. Is that the only one or are there more?

Reply:
I'm not sure, but this is the only example I saw in the json files. Maybe another usage would be (X, Y, Z)

Reply:
Can you run the code and see how many and what units trigger this?